python/pyspark/resultiterable.py

```diff
@@ -15,12 +15,16 @@
 # limitations under the License.
 #
 
-import collections
+try:
+    from collections.abc import Iterable
+except ImportError:
+    from collections import Iterable
```

Comment:
I know this way is more preferred but can we do this by version-checking? That's consistent in the code base at least.

Reply:
Sorry, I was not aware of it.